From 9df5d333304264856465094f9529b414c4c279fb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=81lvaro=20Fern=C3=A1ndez=20Rojas?= <noltari@gmail.com>
Date: Fri, 29 Jul 2016 18:22:37 +0200
Subject: [PATCH] rtl8xxxu: properly detect RTL8188EU devices
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The assumption that all RTL8188EU have chip cut >= C is wrong.
However, RTL8188EU devices can be easily differentiated from RTL8188CU devices
relying on TX report capbility.

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>
Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
---
 drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c
+++ b/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c
@@ -1683,7 +1683,7 @@ static int rtl8xxxu_identify_chip(struct
 		}
 		priv->has_wifi = 1;
 	} else {
-		if (priv->chip_cut >= 2) {
+		if (priv->fops->has_tx_report) {
 			sprintf(priv->chip_name, "8188EU");
 			priv->rf_paths = 1;
 			priv->rx_paths = 1;
